### PR TITLE
ref(integrations): GitHub Refresh token if compare_commits hits a 404 

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

File: src/sentry/integrations/github/repository.py [hunk 1]
```diff
@@ -56,22 +56,32 @@ def build_repository_config(self, organization, data):
...
         try:
-            if start_sha is None:
-                res = client.get_last_commits(name, end_sha)
-                return self._format_commits(client, name, res[:20])
-            else:
-                res = client.compare_commits(name, start_sha, end_sha)
-                return self._format_commits(client, name, res['commits'])
+            return eval_commits(client)
+        except ApiError as e:
+            if '404' in e.message:
```
Comment (markstory) [inline src/sentry/integrations/github/repository.py]: Are you able to use `e.code` for checking the status code?

Comment (lauryndbrown) [reply src/sentry/integrations/github/repository.py]: Oh you're totally right! Thanks for the reminder :)

**Comment by lauryndbrown** [comment]
@markstory let me know what you think now.

**Review by markstory — APPROVED** [review]
Looks good to me. Hopefully this helps with  users hitting 404 errors.

**Pull request merged**